lib/db_connection/connection.ex

```diff
@@ -330,12 +330,22 @@ defmodule DBConnection.Connection do
     handle_timeout(s)
   end
 
-  def handle_event(:info, msg, :no_state, %{mod: mod} = s) do
-    Logger.info(fn ->
-      [inspect(mod), ?\s, ?(, inspect(self()), ") missed message: " | inspect(msg)]
-    end)
+  def handle_event(:info, msg, :no_state, %{mod: mod, state: state} = s) do
+    if function_exported?(mod, :handle_info, 2) do
+      case apply(mod, :handle_info, [msg, state]) do
+        {:ok, state} ->
+          pool_update(state, s)
```

Comment:
I copied this from `ping`.

Reply:
I don't think we need to do a pull update because we don't check the connection out. When we do a ping, [we explicitly check it out from the pool](https://github.com/elixir-ecto/db_connection/blob/master/lib/db_connection/holder.ex#L220), but I don't think we want that here. It should just be a regular message that we don't fiddle with.

Reply:
Does it mean we can't update the state from `handle_info/2` or just that we can return the new state without `pool_update/2`?

Reply:
Good question.

I think we either need to not allow state updates or we need to check it out to avoid race conditions. Because the issue is: what happens if we receive this message when the connection is currently checked out? When they are checked out, the state is in the client.

Another option is to check if the connection is checked out or not. If it is checked out, we queue those messages until they are checked in. Then we execute them and check it in. If it is checked in, then we to check them out, execute it, and check them in.

Given you want this exclusively for disconnect, maybe we can get away with no state update for now, but we need to check what happens if we disconnect while checked out. The client will most likely crash (which I assume is what happens nowadays anyway).

Reply:
> Given you want this exclusively for disconnect, maybe we can get away with no state update for now, but we need to check what happens if we disconnect while checked out. The client will most likely crash (which I assume is what happens nowadays anyway).

Right, I'll update it to `handle_info/1` with only two possible return values `:ok | :disconnect`. This would be totally enough for Ch. And yes, since catching disconnects early is best effort anyway, letting client who has already checked out the connection crash is fine.

Reply:
I would probably do `{:disconnect, reason}` but looks good to me!